jctools-core/src/main/java/org/jctools/queues/atomic/MpmcAtomicArrayQueue.java

```diff
@@ -131,48 +199,49 @@ public E poll() {
         // local load of field to avoid repeated loads after volatile reads
         final AtomicLongArray sBuffer = sequenceBuffer;
         final int mask = this.mask;
-
         long cIndex;
         long seq;
         int seqOffset;
         long expectedSeq;
-        long pIndex = -1; // start with bogus value, hope we don't need it
+        // start with bogus value, hope we don't need it
+        long pIndex = -1;
         do {
             cIndex = lvConsumerIndex();
             seqOffset = calcSequenceOffset(cIndex, mask);
             seq = lvSequence(sBuffer, seqOffset);
             expectedSeq = cIndex + 1;
-            if (seq < expectedSeq) { // slot has not been moved by producer
-                if (cIndex >= pIndex && // test against cached pIndex
-                    cIndex == (pIndex = lvProducerIndex())) { // update pIndex if we must
+            if (seq < expectedSeq) {
+                // slot has not been moved by producer
+                if (// test against cached pIndex
+                cIndex >= pIndex && cIndex == (pIndex = lvProducerIndex())) {
                     // strict empty check, this ensures [Queue.poll() == null iff isEmpty()]
                     return null;
                 } else {
-                    seq = expectedSeq + 1; // trip another go around
+                    // trip another go around
+                    seq = expectedSeq + 1;
                 }
             }
-        } while (seq > expectedSeq || // another consumer beat us to it
-                !casConsumerIndex(cIndex, cIndex + 1)); // failed the CAS
-
+        } while (// another consumer beat us to it
+        seq > expectedSeq || // failed the CAS
+        !casConsumerIndex(cIndex, cIndex + 1));
         final int offset = calcElementOffset(cIndex, mask);
         final E e = lpElement(buffer, offset);
         soElement(buffer, offset, null);
-        soSequence(sBuffer, seqOffset, cIndex + mask + 1);// i.e. seq += capacity
+        // i.e. seq += capacity
+        soSequence(sBuffer, seqOffset, cIndex + mask + 1);
         return e;
     }
-    
+
     @Override
     public E peek() {
         long cIndex;
         E e;
         do {
             cIndex = lvConsumerIndex();
             // other consumers may have grabbed the element, or queue might be empty
-            e = lpElement(calcElementOffset(cIndex));
```

Comment:
This was just missed when I homogenised the atomic queues, the generator has made it consistent.